File: src/main/java/nl/guno/intellij/savechangelisttopatches/SaveChangeListsToPatchesTask.java
package nl.guno.intellij.savechangelisttopatches;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import com.intellij.openapi.progress.ProgressIndicator;
import com.intellij.openapi.progress.Task;
import com.intellij.openapi.project.Project;

class SaveChangeListsToPatchesTask extends Task.Backgroundable {


    SaveChangeListsToPatchesTask(@Nullable Project project) {
        super(project, MessageResources.message("task.saveChangeListsToPatches.title"), false);
    }

    @Override
    public void run(@NotNull ProgressIndicator progressIndicator) {
        new ChangeListsSaver(getProject()).savePatches(false);
    }



}

File: src/main/kotlin/nl/guno/intellij/savechangelisttopatches/SaveChangeListsToPatchesTask.kt
package nl.guno.intellij.savechangelisttopatches

import com.intellij.openapi.progress.ProgressIndicator
import com.intellij.openapi.progress.Task
import com.intellij.openapi.project.Project

internal class SaveChangeListsToPatchesTask(project: Project?)
    : Task.Backgroundable(project, MessageResources.message("task.saveChangeListsToPatches.title"), false) {

    override fun run(progressIndicator: ProgressIndicator) {
        ChangeListsSaver(project).savePatches(false)
    }
}
